sdk/digitaltwins/Azure.DigitalTwins.Core/perf/Azure.DigitalTwins.Core.Perf/Scenarios/QueryDigitalTwins.cs

```diff
@@ -0,0 +1,96 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.DigitalTwins.Core.Perf.Infrastructure;
+using Azure.Identity;
+using Azure.Test.Perf;
+using NUnit.Framework;
+
+namespace Azure.DigitalTwins.Core.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on running queries against digital twins instances.
+    /// </summary>
+    /// <seealso cref="PerfTest{SizeOptions}" />
+    public sealed class QueryDigitalTwins : PerfTest<SizeOptions>
+    {
+        private readonly DigitalTwinsClient _digitalTwinsClient;
+        private readonly string _testId;
+        private readonly long _size;
+        private readonly TimeSpan _delayPeriod = TimeSpan.FromMinutes(1);
+
+        public QueryDigitalTwins(SizeOptions options) : base(options)
+        {
+            _digitalTwinsClient = new DigitalTwinsClient(
+                new Uri(PerfTestEnvironment.Instance.DigitalTwinsUrl),
+                new ClientSecretCredential(
+                    PerfTestEnvironment.Instance.DigitalTwinsTenantId,
+                    PerfTestEnvironment.Instance.DigitalTwinsClientId,
+                    PerfTestEnvironment.Instance.DigitalTwinsClientSecret));
+
+            _size = options.Size;
+            _testId = Guid.NewGuid().ToString().Substring(0, 8);
+        }
+
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            // Global setup code that runs once at the beginning of test execution.
+            // Create the model globally so all tests can take advantage of it.
+            await AdtInstancePopulator.CreateRoomModelAsync(_digitalTwinsClient).ConfigureAwait(false);
```

Comment:
This is a one-time setup for all parallel tests. we only need to do this once per run.